### DISCUSSION TITLE
Review -w hello ( Load error )

### DISCUSSION THREAD

**Discussion by jasondee1992** [kmuto/review#1822] [Q&A] [answered]
reivew-init -w helloを使用しようとすると、このエラーが発生しました。他のRubyGemをインストールする必要がありますか？

C:\Users\******\Desktop\hello>review-init -w hello
<internal:C:/Ruby31-x64/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- webrick (LoadError)
        from <internal:C:/Ruby31-x64/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/review-5.4.0/lib/review/init.rb:311:in `start_webui'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/review-5.4.0/lib/review/init.rb:36:in `execute'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/review-5.4.0/lib/review/init.rb:18:in `execute'
        from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/review-5.4.0/bin/review-init:15:in `<top (required)>'
        from C:/Ruby31-x64/bin/review-init:32:in `load'
        from C:/Ruby31-x64/bin/review-init:32:in `<main>'

**Comment by kmuto** [comment]
おっと。
`gem install webrick` をしてみてください。

-wオプションは使っている人がわずかなようでご意見は貴重です。gemがないときに警告出すようにしますね。

**Reply by jasondee1992** [reply]
インストールが完了し、別のエラーが発生しました。そのようなホストは知られていません。

C:\Users\***\Desktop\hello>gem install webrick
Fetching webrick-1.7.0.gem
Successfully installed webrick-1.7.0
Parsing documentation for webrick-1.7.0
Installing ri documentation for webrick-1.7.0
Done installing documentation for webrick after 4 seconds
1 gem installed

C:\Users\***\Desktop\hello>review-init -w hello
Please access http://<thishost>:18000 from Web browser.
[2022-07-05 09:38:59] INFO  WEBrick 1.7.0
[2022-07-05 09:38:59] INFO  ruby 3.1.2 (2022-04-12) [x64-mingw-ucrt]
⨯ ERROR   Error: getaddrinfo: No such host is known.

**Comment by kmuto** [comment]
手元のLinuxだとデフォルトバインドでいけますが、Windowsではどうもダメのようですね。後でissue化しておきます。
`review-init --bind 127.0.0.1 -w hello` だとどうでしょうか(ローカルホストを明示しています)。

**Reply by jasondee1992** [reply]
申し訳ありませんが、Windowsを使用していることを忘れました 動作していますありがとうございます

**Reply by kmuto** [reply]
よかったです。次のリリースまでに課題を直しておきますね。

**Comment by kmuto** [comment, marked as answer]
本件の当面の回避策

- webrick gemが必要なので、`gem install webrick` を実行する
- Windowsの場合は `review-init --bind 127.0.0.1 -w プロジェクト名` のように明示的にバインドをする必要がある

今後の修正予定
- webrick gemがないときにはインストールを促すメッセージを出してエラー終了させる
- Windowsでのデフォルトのバインドを調査・修正する
